gateway-provider-security-jwt/src/main/java/org/apache/knox/gateway/provider/federation/jwt/filter/JWTFederationFilter.java

```diff
@@ -31,15 +35,22 @@
 import javax.servlet.ServletResponse;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-
 import java.io.IOException;
 import java.nio.charset.StandardCharsets;
 import java.text.ParseException;
 import java.util.Base64;
+import java.util.HashSet;
 import java.util.Locale;
+import java.util.Set;
+import java.util.StringTokenizer;
 
 public class JWTFederationFilter extends AbstractJWTFilter {
 
+  private static final JWTMessages LOGGER = MessagesFactory.get( JWTMessages.class );
+  /* A semicolon separated list of paths that need to bypass authentication */
+  public static final String JWT_UNAUTHENTICATED_PATHS_PARAM = "jwt.unauthenticated.path.list";
+  public static final String DEFAULT_JWT_UNAUTHENTICATED_PATHS_PARAM = "/knoxtoken/api/v1/jwks.json";
```

Comment:
It would be ideal if these constants could be defined elsewhere and shared between HadoopAuthFilter and here instead of being defined in both places.

Reply:
I thought about that but then these auth filters are part of separate packages and extend different filters (one extends Hadoop's and other servlet filter). I don't think there is a suitable common place to house these constants.

Reply:
The HadoopAuthFilter already references the `org.apache.knox.gateway.provider.federation.jwt.filter.JWTFederationFilter`; Couldn't the constants could be defined there, or in AbstractJWTFilter since the module dependency apparently already exists?

Reply:
I moved it to a common util class along with other methods you pointed out.